src/mina-signer/src/field-bigint.ts

```diff
@@ -155,11 +155,21 @@ function checkRange(lower: bigint, upper: bigint, name: string) {
   return (x: bigint) => {
     if (x < lower)
       throw Error(
-        `${name}: inputs smaller than ${lower} are not allowed, got ${x}`
+        `${name}: inputs smaller than ${lower} are not allowed, got ${x}. ${
+          name === 'Scalar'
+            ? 'You may want to use `convertPrivateKeyToBase58WithMod` before using this key.'
+            : ''
```

Comment:
After thinking about it again, checking for `name == 'Scalar'` in this function doesn't seem to be a very precise condition for informing the user about `convertPrivateKeyToBase58WithMod()`. 

the way it's written it will show up on any scalar, not just private keys (for example, scalars that are part of signatures when deserializing a signature).

I would prefer an implementation which catches the error in functions that try to serialize private keys, and if that doesn't work because the input private keys doesn't fall within the range, add the info about `convertPrivateKeyToBase58WithMod()` to the error message